testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AdapterTestExecutionDecider.java

```diff
@@ -35,42 +35,34 @@
 public class AdapterTestExecutionDecider implements TestExecutionDecider {
```

Comment:
This change is fun... hold on tight!

The cache in this method doesn't work properly in a situation we have in this PR, such as `DemoServletsAdapterTest` runs on several app servers, whereas `JettyDemoServletsAdapterTest` runs on a single server which is not mentioned in `DemoServletsAdapterTest`. If `DemoServletsAdapterTest` is executed first, it caches, which methods should run. Let's assume we test with Undertow app server, so the result is `true` for all methods. Next, we execute `JettyDemoServletsAdapterTest` again with Undertow app server. Since, this class is for Jetty, the testsuite shouldn't execute any of these methods (remember, we assumed app server is Undertow). However it tries to do that...

Let me explain why...

The `Method#equals` method uses declaring class (and parameters) to do equality check. Note that the declaring (!!!) method is still the same for both classes - it's `DemoServletsAdapterTest`. This is why it reuses cached entry from `DemoServletsAdapterTest` run instead of re-evaluating whether the test should run or not. Also note, that this caching may introduce more damage that we are not aware of. There are some methods that possible were not executed in our testsuite (or were executed but shouldn't, but if they passed, that's fine). It all depends on the order that Surefire picks.

I decided to remove the cache altogether. What we do in this method is checking a bunch of `if` statements plus traversing test class hierarchy. Both things are blazing fast on modern JVM and CPUs.

Reply:
Maybe I don't get it right but afaik we test single app at the time, so in case we use app-server-undertow any other app server is not tested. The decider should decide if the method is testable or not for this particular container. 

I have nothing against removing the cache completely (as it has minimal impact like you say) in case the decider works as it supposed to.

Reply:
Yes, that's the case. The tricky part is that by default we executed with app server Undertow. However, `JettyDemoServletsAdapterTest` doesn't support it. That's why this tests shouldn't be executed but Surefire was trying to run it. I believe we didn't have similar situation before (Undertow is supported by all the adapters, Jetty is a bit special one).

Yes, I also checked the logs. They look fine. Thanks a lot for the comment!